### PR TITLE
Changes to list options edit

### PR DESCRIPTION
**Pull request by yurige-matrix**
Fixed some bugs and made divide for list.

below added by bradymiller:
Can demo/test this PR in following Up For Grabs demo:
https://www.open-emr.org/wiki/index.php/Development_Demo#Xi_-_Up_For_Grabs_Demo

### CODE REVIEW

**Comment by yurige-matrix** [comment]
Hi @bradymiller . I'm waiting for you when you will review the code

**Comment by bradymiller** [comment]
Plan to review/test this over next couple days.

**Comment by yurige-matrix** [comment]
Hi @yurige-matrix. Many thanks.

File: interface/super/edit_list.php [hunk 1]
```diff
@@ -991,38 +1008,48 @@ function mysubmit() {
...
                 <input type="hidden" name="formaction" id="formaction">
                 <div class="form-group navbar-left">
                     <select name='list_id' class="form-control select-dropdown"
                             id="list_id">
                         <?php
+                        /*
+                         * Keep proper list name (otherwise list name changes according to
+                         * the options shown on the screen).
+                         */
+                        $list_id_container = attr($_GET["list_id_container"]);
```
Comment (bradymiller) [inline interface/super/edit_list.php]: drop the attr(). see my comment I just posted in the prior PR.

File: interface/super/edit_list.php [hunk 2]
```diff
@@ -1209,9 +1276,31 @@ function mysubmit() {
...
+         *    then the list items with edit_options set to 1 will show.
+         */
+
+        /**
+         * In case when we are have a get parametr "list_id_container", we are set manually variable list_id
+         * and range from...to if these is set
+         * adding 1 to from and to limits
+         * And set default order ASC for sort, because we can't use a empty string (sql_limits) for query builder
+         */
+        $sql_limits = 'ASC LIMIT 0, '.$records_per_page;
```
Comment (bradymiller) [inline interface/super/edit_list.php]:
```
$sql_limits = 'ASC LIMIT 0, ' . escape_limit($records_per_page);
```

File: interface/super/edit_list.php [hunk 3]
```diff
@@ -1229,26 +1318,19 @@ function mysubmit() {
...
                 writeITLine(array());
             }
         } else {
-            /*
-             *  Add edit options to show or hide in list management
-             *   If the edit_options setting of the main list entry is set to 0,
-             *    then none of the list items will show.
-             *   If the edit_options setting of the main list entry is set to 1,
-             *    then the list items with edit_options set to 1 will show.
-             */
```
Comment (bradymiller) [inline interface/super/edit_list.php]: Is this supposed to be removed?

File: interface/super/edit_list.php (hunk 1 above)
Comment (yurige-matrix) [reply interface/super/edit_list.php]: Hi @bradymiller . Thank you, erased.

File: interface/super/edit_list.php [hunk 4]
```diff
@@ -1229,26 +1318,19 @@ function mysubmit() {
...
             $res = sqlStatement("SELECT lo.*
-                         FROM list_options as lo
-                         right join list_options as lo2 on lo2.option_id = lo.list_id AND lo2.list_id = 'lists' AND lo2.edit_options = 1
+                         FROM list_options AS lo
+                         RIGHT JOIN list_options AS lo2 ON lo2.option_id = lo.list_id AND lo2.edit_options = 1
```
Comment (bradymiller) [inline interface/super/edit_list.php]: looks like the change above is reverting a recent bug fix

File: interface/super/edit_list.php (hunk 2 above)
Comment (yurige-matrix) [reply interface/super/edit_list.php]: Fixed

File: interface/super/edit_list.php [hunk 5]
```diff
@@ -1342,6 +1424,30 @@ class="btn btn-default btn-save"><?php echo xlt('Save'); ?></button>
             DeleteList(this);
         });
 
+        var totalRecords = '<?php echo $res->_numOfRows;?>';
```
Comment (bradymiller) [inline interface/super/edit_list.php]:
```
var totalRecords = '<?php echo attr($res->_numOfRows); ?>';
```

File: interface/super/edit_list.php [hunk 6]
```diff
@@ -1342,6 +1424,30 @@ class="btn btn-default btn-save"><?php echo xlt('Save'); ?></button>
...
+        var totalRecordDiv = $('#total-record');
+        if( totalRecordDiv ){
+            totalRecordDiv.text("<?php echo xlt("Total records"); ?>: <?php echo $res->_numOfRows;?> <?php echo xlt("items"); ?>");
```
Comment (bradymiller) [inline interface/super/edit_list.php]:
```
totalRecordDiv.text("<?php echo xlt("Total records"); ?>: <?php echo attr($res->_numOfRows); ?> <?php echo xlt("items"); ?>");
```

File: interface/super/edit_list.php [hunk 7]
```diff
@@ -1342,6 +1424,30 @@ class="btn btn-default btn-save"><?php echo xlt('Save'); ?></button>
...
+        }
+
+        var urlFull    = new URL($(location).attr('href'));
+        var listIdCont = urlFull.searchParams.get('list_id_container');
+
+        console.log("list_from: " + listIdCont  );
+        if( totalRecords >= <?php echo $records_per_page;?> || listIdCont != null ) {
```
Comment (bradymiller) [inline interface/super/edit_list.php]: if( totalRecords >= <?php echo attr($records_per_page); ?> || listIdCont != null ) {

File: interface/super/edit_list.php (hunk 3 above)
Comment (yurige-matrix) [reply interface/super/edit_list.php]: You're right. Clean code is cool.

File: interface/super/edit_list.php (hunk 4 above)
Comment (yurige-matrix) [reply interface/super/edit_list.php]: Whats fix?

File: interface/super/edit_list.php (hunk 5 above)
Comment (yurige-matrix) [reply interface/super/edit_list.php]: Fixed

File: interface/super/edit_list.php (hunk 6 above)
Comment (yurige-matrix) [reply interface/super/edit_list.php]: Fixed

File: interface/super/edit_list.php (hunk 7 above)
Comment (yurige-matrix) [reply interface/super/edit_list.php]: Fixed

**Comment by bradymiller** [comment]
hi @yurige-matrix , Some odd stuff I notes on testing:

If I go to the Religious Affiliation list, it shows the to as blank and states it is showing 40 items. Is there a way for it to show something like:
Showing items 1-40 of 120 (or 10-30 of 120 etc.)

Also, if I choose 1 to 20, then it only shows 20 items in the list and doesn't give option to make a custom range.

-brady

**Comment by yurige-matrix** [comment]
Hi @bradymiller . Fixed.

**Comment by bradymiller** [comment]
Can demo/test this PR in following Up For Grabs demo:
https://www.open-emr.org/wiki/index.php/Development_Demo#Xi_-_Up_For_Grabs_Demo

**Comment by bradymiller** [comment]
hi @yurige-matrix ,
Just did some quick testing and this feature is looking really nice. I think one last thing is to populate the to entry with 40 (or whatever the default upper size is) in the cases where the box is visible. For example, when I open the following large list, get this:
image
(if there was a 40 in the second blank box, then it would make more sense)
-brady

**Comment by bradymiller** [comment]
Actually, the above screen is a bad example, here's a better one after opening the religious screen:
https://seven.openemr.io/d/openemr/interface/main/tabs/main.php?token_main=NRHjy2O6zQ2ngVc9nZy87yIx9SQWQzH9Ovkw%2BeJpAxU%3D

**Comment by bradymiller** [comment]
As an aside, the first screenshot above shows a bug that happens when changing the numbers and changing lists. Need to ensure all the counter stuff resets when change to a new list.

**Comment by sjpadgett** [comment]
I still say for a better user experience, these long lists begs for a search. The search can be as simple as a like across all fields. Anything to narrow down to the item in question because chances are if a user is here they are looking for a specific item.
I'll add later if author doesn't want to do.

**Comment by yurige-matrix** [comment]
Hi @bradymiller . It is very good that the function will be useful. What are the counters?

**Comment by yurige-matrix** [comment]
Hi @bradymiller . Fixed display of the upper limit of the filter.

**Comment by yurige-matrix** [comment]
Hi @bradymiller ,  I'm waiting for you when you will review the code

**Comment by bradymiller** [comment]
hi @yurige-matrix ,
Just tested very nicely. Plan to do one more look through the code this evening before bringing into codebase. -brady

**Comment by bradymiller** [comment]
@yurige-matrix , so close. Found following bug when looked at Issues Lists:
image

**Comment by yurige-matrix** [comment]
Hi, @bradymiller . Just fixed a bug with sql for "Issue group"

**Comment by bradymiller** [comment]
hi @yurige-matrix , Code looks good and bringing into codebase. Thanks for all the work (and patience) you put into this code improvement! -brady

**Pull request merged**